packages/BetterPhpDocParser/PhpDocManipulator/PhpDocClassRenamer.php

```diff
@@ -59,15 +60,22 @@ private function processAssertChoiceTagValueNode(array $oldToNewClasses, PhpDocI
 
         $callableCallbackArrayItems = $callbackClass->getValues();
         $classNameArrayItemNode = $callableCallbackArrayItems[0];
+        $classNameStringNode = $classNameArrayItemNode->value;
+
+        // array is needed for callable
+        if (! $classNameStringNode instanceof StringNode) {
+            return;
+        }
 
         foreach ($oldToNewClasses as $oldClass => $newClass) {
-            if ($classNameArrayItemNode->value !== $oldClass) {
+            if ($classNameStringNode->value !== $oldClass) {
                 continue;
             }
 
-            $classNameArrayItemNode->value = $newClass;
+            $classNameStringNode->value = $newClass;
```

Comment:
```suggestion
            $classNameStringNode = $newClass;
```

Reply:
ohh I see now, that you changed it like that because of the changed structure of the class.

IMO you can't change existing classes like that.

Reply:
I think it depends if the class is part of the API or internal.

In order to solve the linked issue, I needed a way to keep the knowledge that either a array item key or value is a literal string, so that no particular processing id done on its value, solving the issue.

The solution I came to is providing a new StringNode (maybe rename it to LiteralStringNode to be explicit?) that hold this knowledge, having as value the content of the string, without the quotes.

It's also closer to Doctrine annotation semantic as a string is strictly a string and should remain a string (e.g. `"123"` represent a numeric string, and should be parsed as a `'123'` string for attributes in PHP code).
Doctrine annotation supports integers if we have written it without quotes :)

It's then up to other services/rectors to make use of the inner string value or not depending on their goal.

Maybe there was a simpler solution?